MSAL/test/unit/native_auth/public/error/SignInStartErrorTests.swift

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import XCTest
+@testable import MSAL
+
+final class SignInStartErrorTests: XCTestCase {
+    private var sut: SignInStartError!
+
+    func test_totalCases() {
+        XCTAssertEqual(SignInStartErrorType.allCases.count, 4)
+    }
+
+    func test_identifier() {
+        let sut: [SignInStartError] = [
+            .init(type: .browserRequired),
+            .init(type: .userNotFound),
+            .init(type: .invalidUsername),
+            .init(type: .generalError)
+        ]
+
+        let identifiers = sut.map { $0.identifier }
+        let expectedIdentifiers = SignInStartErrorType.allCases.map { $0.rawValue }
+
+        zip(identifiers, expectedIdentifiers).forEach {
+            XCTAssertEqual($0, $1)
+        }
+    }
+
+    func test_customErrorDescription() {
+        let expectedMessage = "Custom error message"
+        sut = .init(type: .generalError, message: expectedMessage)
+        XCTAssertEqual(sut.errorDescription, expectedMessage)
+    }
+
+    func test_defaultErrorDescription() {
+        sut = .init(type: .generalError)
+        XCTAssertEqual(sut.errorDescription, SignInStartErrorType.generalError.rawValue)
+    }
+
+
+    func test_isBrowserRequired() {
+        sut = .init(type: .browserRequired)
+        
```

Comment:
nit: spacing, same for below tests, also order should be kept for where the `XCTAssertTrue` is like in the other tests classes such as `SignInPasswordStartErrorTests`

Reply:
Done.